interop/tapir/src/main/scala/caliban/interop/tapir/TapirAdapter.scala

```diff
@@ -230,18 +232,38 @@ object TapirAdapter {
     makeHttpUploadEndpoint.serverLogic(logic)
   }
 
-  def makeWebSocketEndpoint[R, E](implicit
+  /**
+   * A codec which expects only text and close frames (all other frames cause a decoding error). Close frames correspond to a `Left`,
+   * while text frames are handled using the given `stringCodec` and wrapped with a `Right`
+   * @return
+   */
+  private implicit def textOrCloseWebSocketFrameEither[A, CF <: CodecFormat](implicit
```

Comment:
We previously relied on a function in Tapir (if I remember properly), what's different in this one?

Reply:
So before it didn't handle any other frames IIRC, and tapir has another method that uses an option instead and converts the `None` into a standard close frame. This is needed in order to handle the websocket error codes that are specified in the transport specification